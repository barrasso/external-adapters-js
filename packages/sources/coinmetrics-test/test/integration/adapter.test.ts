import { mockBurnedSuccess, mockPriceSuccess, mockTotalBurnedSuccess } from './fixtures'
import { SuperTest, Test } from 'supertest'
import { setupExternalAdapterTest, SuiteContext } from './setup'
import { ServerInstance } from '@chainlink/external-adapter-framework'

describe('execute', () => {
  const id = '1'

  const context: SuiteContext = {
    req: null,
    server: async () => {
      process.env['RATE_LIMIT_CAPACITY_SECOND'] = '6'
      process.env['METRICS_ENABLED'] = 'false'
      const server = (await import('../../src')).server
      return server() as Promise<ServerInstance>
    },
  }

  const envVariables = {
    CACHE_ENABLED: 'false',
    API_ENDPOINT: process.env.API_ENDPOINT || 'http://localhost:18081',
    API_KEY: process.env.API_KEY || 'test_key',
  }

  setupExternalAdapterTest(envVariables, context)

  describe('price api', () => {
    const data = {
      id,
      data: {
        base: 'ETH',
        quote: 'USD',
      },
    }

    it('should return success', async () => {
      mockPriceSuccess()

      const response = await (context.req as SuperTest<Test>)
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })

  describe('total-burned api', () => {
    const data = {
      id,
      data: {
        asset: 'eth',
        frequency: '1d',
        pageSize: 10000,
        startTime: '2021-09-20',
        endTime: '2021-09-25',
        endpoint: 'total-burned',
      },
    }

    it('should return success', async () => {
      mockTotalBurnedSuccess()

      const response = await (context.req as SuperTest<Test>)
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })

  describe('burned api', () => {
    const data = {
      id,
      data: {
        asset: 'eth',
        frequency: '1d',
        endpoint: 'burned',
      },
    }

    it('should return success', async () => {
      mockBurnedSuccess()

      const response = await (context.req as SuperTest<Test>)
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
