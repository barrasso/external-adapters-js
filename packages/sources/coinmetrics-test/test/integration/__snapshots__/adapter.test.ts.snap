// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute burned api should return success 1`] = `
Object {
  "providerStatusCode": 200,
  "result": "1580.43730192317682277",
  "statusCode": 200,
}
`;

exports[`execute price api should return success 1`] = `
Object {
  "providerStatusCode": 200,
  "result": 1552.8,
  "statusCode": 200,
}
`;

exports[`execute total-burned api should return success 1`] = `
Object {
  "providerStatusCode": 200,
  "result": "40741.188143386670627111",
  "statusCode": 200,
}
`;
